plugins/faustwp/includes/replacement/functions.php

```diff
@@ -131,3 +131,106 @@ function is_wp_link_ajax_request(): bool {
 		&& ! empty( $_POST['action'] )
 		&& 'wp-link-ajax' === $_POST['action'] );
 }
+
+/**
+ * Get all site URLs for each possible HTTP protocol
+ */
+function faustwp_get_wp_site_urls() {
+
+	$site_url = site_url();
+	$host_url = wp_parse_url( $site_url, PHP_URL_HOST );
+
+	if ( ! is_string( $host_url ) ) {
+		return apply_filters( 'faustwp_get_wp_site_urls', array( $site_url ) );
+	}
+
+	$is_https = substr( $site_url, 0, 6 ) === 'https:';
+
+	return apply_filters(
+		'faustwp_get_wp_site_urls',
+		array(
+			$is_https ? "https://$host_url" : "http://$host_url",
+			$is_https ? "http://$host_url" : "https://$host_url",
+			"//$host_url",
+		)
+	);
+}
+
+/**
+ * Get all media urls based off the available site urls
+ *
+ * @return array
+ */
+function faustwp_get_wp_media_urls() {
+	$site_urls  = faustwp_get_wp_site_urls();
+	$upload_url = faustwp_get_relative_upload_url( $site_urls );
+
+	if ( ! is_string( $upload_url ) ) {
+		return apply_filters( 'faustwp_get_wp_site_media_urls', array() );
+	}
+
+	$media_urls = array();
+	foreach ( $site_urls as $site_url ) {
+		$media_urls[] = $site_url . $upload_url;
+	}
+
+	return apply_filters( 'faustwp_get_wp_site_media_urls', $media_urls );
+}
+
+
+/**
+ * Gets the relative wp-content upload URL.
+ *
+ * @param array<string> $site_urls An array of site URLs.
+ *
+ * @return string The relative upload URL.
+ */
+function faustwp_get_relative_upload_url( $site_urls ) {
+	$upload_dir = wp_upload_dir()['baseurl'];
+
+	foreach ( $site_urls as $site_url ) {
+		if ( false !== strpos( $upload_dir, $site_url ) ) {
```

Comment:
Are we checking if it starts with or contains?

Reply:
Sorry, I mean that `strpos($haystack, $needle) === 0` verifies if it **starts** with and `strpos($haystack, $needle) !== false` that it **contains**.